### PR TITLE
fixing the mounts for the daemonset config to have non subpath mount

### PR DESCRIPTION
**Pull request by mwoodson**

### CODE REVIEW

**Comment by mwoodson** [comment]
@kwoodson 
@sdodson This needs to be cherry-picked into release-3.9

/cherry-pick 3.9

File: roles/openshift_daemonset_config/templates/daemonset.yml.j2 [hunk 1]
```diff
@@ -150,3 +149,7 @@ spec:
       - name: host
         hostPath:
           path: /
+      - hostPath:
+          path: /run/docker.sock
```
Comment (liggitt) [inline roles/openshift_daemonset_config/templates/daemonset.yml.j2]: I expected `/var/run/docker.sock` here

File: roles/openshift_daemonset_config/templates/daemonset.yml.j2 [hunk 2]
```diff
@@ -101,9 +101,8 @@ spec:
           subPath: sys
           name: host
           readOnly: true
-        - mountPath: /var/run/docker.sock
-          subPath: var/run/docker.sock
-          name: host
+        - mountPath: /run/docker.sock
```
Comment (liggitt) [inline roles/openshift_daemonset_config/templates/daemonset.yml.j2]: I expected `/var/run/docker.sock` here

File: roles/openshift_daemonset_config/templates/daemonset.yml.j2 (hunk 1 above)
Comment (kwoodson) [reply roles/openshift_daemonset_config/templates/daemonset.yml.j2]: I did as well but /var/run is a symlink to /run.  

That's why this works.  I'm fine either way.

File: roles/openshift_daemonset_config/templates/daemonset.yml.j2 (hunk 2 above)
Comment (mwoodson) [reply roles/openshift_daemonset_config/templates/daemonset.yml.j2]: same as above.  /var/run is a symlink to /run.

I can change it to /var/run if that's preferred.

Comment (liggitt) [reply roles/openshift_daemonset_config/templates/daemonset.yml.j2]: if it works, I don't care, just noting it wasn't identical

**Pull request merged**